include/seastar/core/metrics_api.hh

```diff
@@ -59,6 +65,23 @@ struct hash<seastar::metrics::impl::labels_type> {
     }
 };
 
+template<>
+struct hash<seastar::metrics::impl::replicated_family_spec> {
+    using argument_type = seastar::metrics::impl::replicated_family_spec;
+    using result_type = ::std::size_t;
+    result_type operator()(argument_type const& spec) const {
+        return std::hash<seastar::sstring>{}(spec.full_name);
```

Comment:
I don't think we'll require registration on multiple destinations. The API allows for it, but it's awkward since I didn't design for it.

Transparent comparators are neat. I was wondering if there's a way to avoid constructing a key when I was writing the look-up code. `std::unordered_multimap<ss::sstring, handle>` would work too. Let me think about it.

Reply:
I don't think the API allows for it, the size of this container is 1:
```c++
    std::unordered_set<replicated_family_spec> metric_set{
      replicated_family_spec{0, "family"}, replicated_family_spec{1, "family"}};
```

Reply:
Yep. You're correct. I think that's misleading for the user as they'd have to be aware of the `equal_to` override in order to understand that the size of this container is one. I've changed the internal storage and the APIs above it to use a `std::unordered_map<sstring, int>`. This better reflects the fact that there may be only one destination for each metric family.

Reply:
Removed the limitation. See reply to other related comment [here](https://github.com/redpanda-data/seastar/pull/27#discussion_r918876089).